apps/mcp-ui-poc/server/src/elements/button.ts

```diff
@@ -8,6 +8,8 @@ export interface ButtonElementArgs {
   isDisabled?: boolean;
   type?: "button" | "submit" | "reset";
   ariaLabel?: string;
+  /** Optional intent to emit when button is pressed */
+  intent?: Intent;
```

Comment:
@ByronDWall It is powerful, if not risky, to give the agent complete control over the intent.

One of the issues I've been grappling with lately is _how much control_ we exert over these UI templates when it comes to statefulness, that is, what priority data do _we_ attach to handlers (product name, id, sku, etc), and I chose to lean into the inverse of this to give agents more control (continued below)